### PR TITLE
Introduce "Entity" struct, replacing "Interactables"

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: include/constants/interactables.h [hunk 1]
```diff
@@ -72,7 +72,7 @@
 #define IA__ICE_PARADISE__HALFPIPE__START          68
 #define IA__069                                    69
 #define IA__070                                    70
-#define IA__071                                    71 // Interactable 071 calls the code of IA__044
+#define IA__071                                    71 // MapEntity 071 calls the code of IA__044
```
Comment (freshollie) [inline include/constants/interactables.h:75]: I guess these comments don't need to be replaced?

File: include/game/interactables_1/bouncy_bar.h [hunk 2]
```diff
@@ -2,9 +2,9 @@
 #define GUARD_INTERACTABLE_BOUNCY_BAR_H
 
 #include "global.h"
-#include "game/interactable.h"
+#include "game/entity.h"
 
-void initSprite_Interactable_BouncyBar(Interactable *ia, u16 spriteRegionX,
```
Comment (freshollie) [inline include/game/interactables_1/bouncy_bar.h:7]: Are we gonna replace all `ia` with `me` eventually?

File: include/constants/interactables.h (hunk 1 above)
Comment (JaceCear) [reply include/constants/interactables.h:75]: Ah damn... I used case-sensitive auto-replace with full names...

File: include/game/interactables_1/bouncy_bar.h (hunk 2 above)
Comment (JaceCear) [reply include/game/interactables_1/bouncy_bar.h:7]: If with 'eventually' you mean ASAP, then yes!

Though the category "interactable" should still exist, like it does for "enemy" - just not the type.

File: ldscript.txt [hunk 3]
```diff
@@ -161,7 +161,7 @@ SECTIONS
         build/sa2/src/game/interactables_2/notification_ring_bonus.o(.text);
         build/sa2/src/game/interactables_2/sky_canyon_init.o(.text);
         build/sa2/src/game/interactables_2/105.o(.text);
-        build/sa2/src/game/interactables_2/note_particle.o(.text);
```
Comment (freshollie) [inline ldscript.txt:164]: For now I am following the linker for where things should be placed. Based on that this isn't in the music plant folder (as it's not grouped with the other music plant stuff)

**Review by freshollie — APPROVED** [review]

File: ldscript.txt (hunk 3 above)
Comment (freshollie) [reply ldscript.txt]: I've undone this so I can merge for my changes

**Pull request merged**